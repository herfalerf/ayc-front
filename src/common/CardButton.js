import React from "react";
import { Link } from "react-router-dom";
import { makeStyles } from "@material-ui/core/styles";
import {
  Card,
  CardActionArea,
  CardContent,
  Typography,
  Icon,
  Grid,
} from "@material-ui/core";
import ic_arrow_small_blue from "./images/ic_arrow_small_blue.svg";

// Reusable card button component, will make a card button which accepts an icon, header text, bodytext a route and an onClick function
//

const useStyles = makeStyles((theme) => ({
  root: {
    maxWidth: 850,
  },
  cardAction: {},
  mainIcon: {
    height: 60,
  },
  arrowIcon: {
    height: 25,
  },
}));

const CardButton = ({ icon, alt, headerText, bodyText, route, onClick }) => {
  const classes = useStyles();
  return (
    <Card className={classes.root}>
      <CardActionArea
        className={classes.cardAction}
        component={Link}
        to={route}
        onClick={onClick}
      >
        <CardContent>
          <Grid container justifyContent="space-between" alignItems="center">
            <Grid item xs={2}>
              <Icon>
                <img src={icon} alt={alt} className={classes.mainIcon} />
              </Icon>
            </Grid>
            <Grid item xs={8}>
              <Typography variant="h5" component="h5" color="inherit">
                {headerText}
              </Typography>
              <Typography variant="body2" color="inherit">
                {bodyText}
              </Typography>
            </Grid>
            <Grid item xs={1}>
              <Icon>
                <img
                  src={ic_arrow_small_blue}
                  alt=""
                  className={classes.arrowIcon}
                />
              </Icon>
            </Grid>
          </Grid>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

export default CardButton;
