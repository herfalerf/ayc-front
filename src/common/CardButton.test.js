import React from "react";
import { render } from "@testing-library/react";
import CardButton from "./CardButton";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <CardButton route="/" />
    </MemoryRouter>
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <CardButton route="/" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
