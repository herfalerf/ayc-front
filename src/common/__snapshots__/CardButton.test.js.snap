// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiCardActionArea-root makeStyles-cardAction-6"
      href="/"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiCardContent-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-content-xs-space-between"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
              <img
                class="makeStyles-mainIcon-7"
              />
            </span>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
          >
            <h5
              class="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
            />
            <p
              class="MuiTypography-root MuiTypography-body2 MuiTypography-colorInherit"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
          >
            <span
              aria-hidden="true"
              class="material-icons MuiIcon-root"
            >
              <img
                alt=""
                class="makeStyles-arrowIcon-8"
                src="ic_arrow_small_blue.svg"
              />
            </span>
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </div>
</DocumentFragment>
`;
